contracts/auction/ClockAuction.sol

```diff
@@ -103,9 +104,13 @@ contract ClockAuction is AuctionRelated {
         require(_bidAmount >= priceInETH,
             "your offer is lower than the current price, try again with a higher one.");
 
-        // assure that this get ring back from tokenVendor
-        require(tokenVendor.buyToken.value(_bidAmount)(address(this)));
+        uint refund = _bidAmount - priceInETH;
+        if (refund > 0) {
+            _buyer.transfer(refund);
+        }
 
+        // assure that this get ring back from tokenVendor
+        require(tokenVendor.buyToken.value(priceInETH)(address(this)));
```

Comment:
later, after migrate to bancor, should add the amount of RING to buy(...) as minTokenReturn.